### ISSUE TITLE
ListItem sort does not work in the unified builtins branch

### ISSUE THREAD

**Issue by liyili2** [kframework/k-legacy#524]
The c-semantics need the following rule to be able to do some jobs. 

```
module TEST

  syntax ListItem ::= "next"

  rule next ListItem(A:K) => .List

endmodule
```

However, the current parser will complain about this rule and show something like:

[Error] Critical: Parse error: Syntax error near unexpected character 'L'
    File: /home/qangel/project/k-unified/bin/test.k
    Location: (5,13,5,13)

If I change the rule to the following rule:

```
module TEST

  syntax List ::= "next"

  rule next ListItem(A:K) => .List

endmodule
```

The compiler will complain that:

[Error] Critical: unexpected left-hand side data structure format; expected
elements and at most one variable
'_List_('next(),, 'ListItem(A:K ))
    File: /home/qangel/project/k-unified/bin/test.k
    Location: (5,3,5,34)
    Compilation Phase: Compile collections to internal K representation

If I try this following rule:

```
module TEST

  syntax List ::= "next"

  rule <k> next ListItem(A:K) => .List </k>

endmodule
```

The compiler will still have the same output as:

[Error] Critical: unexpected left-hand side data structure format; expected
elements and at most one variable
'_List_('next(),, 'ListItem(A:K ))
    File: /home/qangel/project/k-unified/bin/test.k
    Location: (5,3,5,43)
    Compilation Phase: Compile collections to internal K representation

Basically, I understand I cannot match something in the middle of a list, but it is very sad to know that I cannot declare a ListItem and cannot match something in the end of a list.

**Comment by yilongli** [comment]
Why not write `ListItem(next)`? Just use `ListItem` as a wrapper.

**Comment by liyili2** [comment]
I think I can do that, but I do not like this idea, because now the rules will depend on the content of ListItem(A). If other rules match the whole ListItem(A:K) block and do something else, then it will be very bad, because it is very hard for me to detect whether or not the C semantics want to include mathing the ListItem(next) case or exclude the ListItem(next) case in other contexts.

**Comment by liyili2** [comment]
In addition, it is very hard for a designer to design a language, since they must put a condition to say that when "A =/= next" in every place they want to match the "ListItem(A)" block and this is just one such case. If the user define 100 this kind of constructor, then they must put 100 inequality conditions in every rule that they want to match  the "ListItem(A)" .
